src/test/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImplTest.java

```diff
@@ -0,0 +1,369 @@
+        when(skillRepository.findAll()).thenReturn(skills);
+        StaffWorkerSkillResponse staffWorkerSkillResponse = staffRefDataServiceImpl.getServiceSkills();
+
+        assertThat(staffWorkerSkillResponse).isNotNull();
+
+        List<ServiceSkill> serviceSkills = staffWorkerSkillResponse.getServiceSkills();
+
+        assertThat(serviceSkills).isNotNull();
+
+        ServiceSkill serviceSkill = serviceSkills.get(0);
+
+        assertThat(serviceSkill.getId()).isEqualTo("BBA3");
+
+        SkillDTO skillDTO = serviceSkill.getSkills().get(0);
+
+        assertThat(skillDTO.getSkillId()).isEqualTo(1L);
+        assertThat(skillDTO.getSkillCode()).isEqualTo("A1");
+        assertThat(skillDTO.getDescription()).isEqualTo("desc1");
+        assertThat(skillDTO.getUserType()).isEqualTo("user_type1");
+
+    }
+
+    @Test
+    void should_return_empty_list_of_service_skills() {
+        List<Skill> skills = new ArrayList<>();
+        when(skillRepository.findAll()).thenReturn(skills);
+        StaffWorkerSkillResponse staffWorkerSkillResponse = staffRefDataServiceImpl.getServiceSkills();
+
+        assertThat(staffWorkerSkillResponse).isNotNull();
+
+        List<ServiceSkill> serviceSkills = staffWorkerSkillResponse.getServiceSkills();
+        assertThat(serviceSkills).isNotNull();
+
+        assertThat(serviceSkills).isEmpty();
+    }
+
+    @Test
+    void should_return_null_list_of_service_skills() {
+        List<Skill> skills = null;
+        when(skillRepository.findAll()).thenReturn(skills);
+        StaffWorkerSkillResponse staffWorkerSkillResponse = staffRefDataServiceImpl.getServiceSkills();
+
+        assertThat(staffWorkerSkillResponse).isNotNull();
+
+        List<ServiceSkill> serviceSkills = staffWorkerSkillResponse.getServiceSkills();
+        assertThat(serviceSkills).isNotNull();
+
+        assertThat(serviceSkills).isEmpty();
+    }
+
+    @Test
+    void should_return_case_worker_profile_with_status_code_200() {
+
+        var pageRequest =
+                validateAndBuildPagination(20, 1,
+                        20, 1);
+
+        String searchString = "cwr";
+        ArrayList<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        CaseWorkerProfile caseWorkerProfile = buildCaseWorkerProfile();
+        caseWorkerProfiles.add(caseWorkerProfile);
+        Page<CaseWorkerProfile> pages = new PageImpl<>(caseWorkerProfiles);
+
+        when(caseWorkerProfileRepo.findByFirstNameOrLastName(searchString.toLowerCase(), pageRequest))
+                .thenReturn(pages);
+        ResponseEntity<List<SearchStaffUserResponse>> responseEntity =
+                staffRefDataServiceImpl.retrieveStaffUserByName(searchString, pageRequest);
+        assertEquals(200, responseEntity.getStatusCodeValue());
+
+        List<SearchStaffUserResponse> searchResponse =
+                responseEntity.getBody();
+        assertThat(responseEntity.getBody()).isNotNull();
+        validateSearchStaffUserResponses(searchResponse);
```

Comment:
check  response header has total records and its value.

Reply:
done